earthaccess/auth.py

```diff
@@ -116,70 +117,18 @@ def _set_earthdata_system(self, system: System) -> None:
         # Maybe all these predefined URLs should be in a constants.py file
-        self.EDL_GET_TOKENS_URL = f"https://{self.system.edl_hostname}/api/users/tokens"
-        self.EDL_GENERATE_TOKENS_URL = (
-            f"https://{self.system.edl_hostname}/api/users/token"
-        )
-        self.EDL_REVOKE_TOKEN = (
-            f"https://{self.system.edl_hostname}/api/users/revoke_token"
+        self.EDL_FIND_OR_CREATE_TOKEN_URL = (
+            f"https://{self.system.edl_hostname}/api/users/find_or_create_token"
         )
 
         self._eula_url = (
             f"https://{self.system.edl_hostname}/users/earthaccess/unaccepted_eulas"
         )
         self._apps_url = f"https://{self.system.edl_hostname}/application_search"
 
+    @deprecated("Refresh tokens has been deprecated and is not necessary anymore.")
     def refresh_tokens(self) -> bool:
-        """Refresh CMR tokens.
-
-        Tokens are used to do authenticated queries on CMR for restricted and early access datasets.
-        This method renews the tokens to make sure we can query the collections allowed to our EDL user.
-        """
-        if len(self.tokens) == 0:
-            resp_tokens = self._generate_user_token(
-                username=self.username, password=self.password
-            )
-            if resp_tokens.ok:
-                self.token = resp_tokens.json()
-                self.tokens = [self.token]
-                logger.debug(
-                    f"earthaccess generated a token for CMR with expiration on: {self.token['expiration_date']}"
-                )
-                return True
-            else:
-                return False
-        if len(self.tokens) == 1:
-            resp_tokens = self._generate_user_token(
-                username=self.username, password=self.password
-            )
-            if resp_tokens.ok:
-                self.token = resp_tokens.json()
-                self.tokens.extend(self.token)
-                logger.debug(
-                    f"earthaccess generated a token for CMR with expiration on: {self.token['expiration_date']}"
-                )
-                return True
-            else:
-                return False
-
-        if len(self.tokens) == 2:
-            resp_revoked = self._revoke_user_token(self.token["access_token"])
-            if resp_revoked:
-                resp_tokens = self._generate_user_token(
-                    username=self.username, password=self.password
-                )
-                if resp_tokens.ok:
-                    self.token = resp_tokens.json()
-                    self.tokens[0] = self.token
-                    logger.debug(
-                        f"earthaccess generated a token for CMR with expiration on: {self.token['expiration_date']}"
-                    )
-                    return True
-                else:
-                    logger.info(resp_tokens)
-                    return False
-
-        return False
+        return self.authenticated
```

Comment:
Please put the docstring back in place. When we deprecate something, we still want to keep its docstring in place.